mayastor/src/core/env.rs

```diff
@@ -199,6 +213,9 @@ impl Default for MayastorEnvironment {
     fn default() -> Self {
         Self {
             config: None,
+            enable_grpc: false,
+            grpc_addr: "None".into(),
```

Comment:
The default should be really the POD IP address taken from environment variable MY_POD_IP. And 127.0.0.1 can be used if MY_POD_IP is not set and cmd line argument not specified.

Reply:
Structopt claims you can write ``default_value`` without the ``="whatever" `` in the declaration, and then it will pick up the value from the Default impl.  In practice, I could not get this to work, and had to put the default in the structopt declaration.  However, the Default impl still needed a value.  That is why the "127.0.0.1" ended up in two places.

Reply:
Just an idea, but perhaps we could avoid enable_grpc and strange "None" default value by having an optional member in ms env that is a tuple `(address, port)`. We could call it `grpc_endpoint` and it would be simply set to `None` if grpc is not used. So host part would be `grpc_endpoint.0` and port `grpc_endpoint.1`. It also eliminates the case when enable_grpc=true but grpc_addr remains "None" because it was not set.

Reply:
No, no.  The value is always set and never remains "None".  The "None" is overwritten by the value from structopt unconditionally.  As for merging the enable flag with the presence or not of the socket settings, I think that will confuse the logic - the flag is set by the client code that uses env.  The socket settings are set by the end user.

Reply:
I will try to explain better what I mean:

in struct MayastorEnvironment we would have:
```
+       grpc_endpoint: (String, String),
```
In impl Default for MayastorEnvironment:
```
+       grpc_endpoint: None,
```
and in MayastorEnvironment::new:
```
   Self {
        ...
+      grpc_endpoint: (args.addr, args.port),
        ..Default::default()
   }
```
I find this as a more natural way how to express what we want in rust. First, we don't need a special placeholder string "None" if rust already has a special value to denote empty values (Optional). Second, I don't need three independent variables in mayastor env where only a few combinations of them make sense - so less error-prone I would say. Not sure how that seems to you guys?

Reply:
"None" is not a valid value - it is not used to signify a not-specified address, unlike an Option which is None.  Also, in your scheme, how would the client code enable or disable the grpc?